quixstreams/models/topics/manager.py

```diff
@@ -201,147 +202,144 @@ def register(self, topic: Topic) -> Topic:
 
         topic = self._finalize_topic(
             Topic(
-                name=self._internal_name("changelog", topic_name, store_name),
+                name=self._internal_name("changelog", stream_id, store_name),
                 key_serializer="bytes",
                 value_serializer="bytes",
                 key_deserializer="bytes",
                 value_deserializer="bytes",
                 create_config=config,
             )
         )
-        self._changelog_topics.setdefault(topic_name, {})[store_name] = topic
+        if topic.broker_config.num_partitions != config.num_partitions:
+            raise TopicConfigurationMismatch(
+                f'Invalid partition count for the changelog topic "{topic.name}": '
+                f"expected {config.num_partitions}, "
+                f'got {topic.broker_config.num_partitions}"'
+            )
+
+        self._changelog_topics.setdefault(stream_id, {})[store_name] = topic
         return topic
 
-    def validate_all_topics(self):
-        """
-        Validates that all topics have ".broker_config" set
-        and changelog topics have correct numbers of partitions and replication factors.
-
-        Issues are pooled and raised as an Exception once inspections are complete.
-        """
-        logger.info("Validating Kafka topics are configured correctly")
-        for source_topic in self.non_changelog_topics.values():
-            # For any changelog topics, validate the amount of partitions and
-            # replication factor match with the source topic
-            source_config = source_topic.broker_config
-            for changelog_topic in self.changelog_topics.get(
-                source_topic.name, {}
-            ).values():
-                changelog_config = changelog_topic.broker_config
-                if changelog_config.num_partitions != source_config.num_partitions:
-                    raise TopicConfigurationMismatch(
-                        f'changelog topic "{changelog_topic.name}" partition count '
-                        f'does not match its source topic "{source_topic.name}": '
-                        f"expected {source_config.num_partitions}, "
-                        f'got {changelog_config.num_partitions}"'
-                    )
-                if (
-                    changelog_config.replication_factor
-                    != source_config.replication_factor
-                ):
-                    raise TopicConfigurationMismatch(
-                        f'changelog topic "{changelog_topic.name}" replication factor '
-                        f'does not match its source topic "{source_topic.name}": '
-                        f"expected {source_config.replication_factor}, "
-                        f'got {changelog_config.replication_factor}"'
-                    )
-
-        logger.info("Kafka topics validation complete")
+    @classmethod
+    def derive_topic_config(cls, *topics: Topic) -> TopicConfig:
```

Comment:
Same here, maybe just accept a single `topics` argument.
```suggestion
    def derive_topic_config(cls, topics: FrozenSet[Topic]) -> TopicConfig:
```